File: java/src/main/java/com/experian/dvs/client/email/validate/ValidateResult.java
package com.experian.dvs.client.email.validate;

import com.experian.dvs.client.common.ResponseError;
import com.experian.dvs.client.email.EmailConfidence;
import com.experian.dvs.client.server.email.RestApiEmailMetadata;
import com.experian.dvs.client.server.email.RestApiEmailValidateResponse;
import com.experian.dvs.client.server.email.RestApiEmailValidateResult;

import java.util.List;
import java.util.Optional;

public class ValidateResult {

    private final ResponseError error;
    private final EmailConfidence confidence;
    private final List<String> didYouMean;
    private final VerboseOutput verboseOutput;
    private final DomainType domainType;

    public ValidateResult(RestApiEmailValidateResponse apiResponse) {
        this.error = apiResponse.getError() != null ? new ResponseError(apiResponse.getError()) : null;

        final RestApiEmailValidateResult result = apiResponse.getResult();
        if (result != null) {
            this.confidence = result.getConfidence() != null ? EmailConfidence.fromName(result.getConfidence()) : null;
            this.didYouMean = result.getDidYouMean() != null ? List.copyOf(result.getDidYouMean()) : List.of();
            this.verboseOutput = result.getVerboseOutput() != null ? VerboseOutput.fromName(result.getVerboseOutput()) : null;
        } else {
            this.confidence = null;
            this.didYouMean = List.of();
            this.verboseOutput = null;

        }
        final RestApiEmailMetadata metadata = apiResponse.getMetadata();
        if (metadata != null) {
            this.domainType = metadata.getDomainDetail() != null && metadata.getDomainDetail().getType() != null ? DomainType.fromName(metadata.getDomainDetail().getType()) : null;
        } else {
            this.domainType = null;
        }
    }

    public Optional<ResponseError> getError() {
        return Optional.ofNullable(this.error);
    }

    public EmailConfidence getConfidence() {
        return confidence;
    }

    public List<String> getDidYouMean() {
        return didYouMean;
    }

    public VerboseOutput getVerboseOutput() {
        return verboseOutput;
    }

    public DomainType getDomainType() {
        return domainType;
    }
}

File: csharp/DVSClient/DVSClient/email/validate/ValidateResult.cs
using DVSClient.Common;
using DVSClient.Server.Email;

namespace DVSClient.Email.Validate
{
    public class ValidateResult
    {
        public ResponseError? Error { get; }
        public EmailConfidence? Confidence { get; }
        public IEnumerable<string> DidYouMean { get; }
        public VerboseOutput? VerboseOutput { get; }
        public DomainType? DomainType { get; }

        public ValidateResult(RestApiEmailValidateResponse apiResponse)
        {
            Error = apiResponse.Error != null ? new ResponseError(apiResponse.Error) : null;

            var result = apiResponse.Result;
            if (result != null)
            {
                Confidence = result.Confidence?.GetEnumValueFromJsonName<EmailConfidence>();
                DidYouMean = result.DidYouMean != null ? new List<string>(result.DidYouMean) : new List<string>();
                VerboseOutput = result.VerboseOutput?.GetEnumValueFromJsonName<VerboseOutput>();
            }
            else
            {
                Confidence = default;
                DidYouMean = new List<string>();
                VerboseOutput = default;
            }

            var metadata = apiResponse.Metadata;
            if (metadata != null)
            {
                DomainType = metadata.DomainDetail?.Type?.GetEnumValueFromJsonName<DomainType>();
            }
            else
            {
                DomainType = default;
            }
        }
    }
}
